developer_manual/app/custom-cache-backend.rst

```diff
@@ -0,0 +1,246 @@
+
+Updater
+-------
+
+Another cache related system, which developers should be aware of when working with custom caches, is the updater. 
+The updater (``\OC\Files\Cache\Updater``) is responsible for updating the cache when any change is mode from inside ownCloud.
+It will call either the scanner or the cache of a storage to make the required changes.
+The updater can **not** be overwritten by storage backends.
+
++------------------------+----------------------------------------------------------------------------------------------+
+| Method                 | Description                                                                                  |
++========================+==============================================================================================+
+| ``checkUpdate($path)`` | Checks if a file or folder has been changed externally. If it has, then update the cache and |
+|                        | return ``true``. Otherwise, it returns ``false``.                                            |
++------------------------+----------------------------------------------------------------------------------------------+
+| ``cleanFolder($path)`` | Checks a folder for any child entries that are no longer in the storage. It should be        |
+|                        | called automatically by ``checkUpdate`` if that method detects an update.                    |
++------------------------+----------------------------------------------------------------------------------------------+
+
+An app or an administrator can also change the watcher behavior, by setting it's policy through ``setPolicy($policy)``. 
+This method can take the following values:
+
++---------------------------+-------------------------------------------------------------------------------+
+| Method                    | Description                                                                   |
++===========================+===============================================================================+
+| ``Watcher::CHECK_NEVER``  | Don’t check for any external change (recommended if you're certain no outside |
+|                           | changes will be made).                                                        |
++---------------------------+-------------------------------------------------------------------------------+
+| ``Watcher::CHECK_ONCE``   | Check each path for updates at most once during a request (default).          |
++---------------------------+-------------------------------------------------------------------------------+
+| ``Watcher::CHECK_ALWAYS`` | Check for external changes any number of times during a request, discouraged, |
+|                           | mostly useful for unit tests.                                                 |
++---------------------------+-------------------------------------------------------------------------------+
```

Comment:
Looks like a lot of Watcher information duplicated in the Updater tables above???